### PR TITLE
[terra-form-textarea] add `aria-invalid` field

### PR DESCRIPTION
**Pull request by razvanluca**
### Summary

Added `aria-invalid` field to mark what field is invalid for screen readers

**What was changed:**
Added `aria-invalid` field to [terra-form-textarea]

**Why it was changed:**
Added `aria-invalid` field to mark what field is invalid for screen readers

### Testing

This change was tested using:

- [ ] WDIO
- [x] Jest
- [x] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

https://github.com/cerner/terra-core/assets/13691565/12452937-9ebb-44fa-9e27-3f88704752d2

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [x] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9356

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-form-textarea/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Changelog
 
 ## Unreleased
+* Added `aria-invalid` field to Textarea component and changed existing example so it can be tested
```
Comment (sdadn) [inline packages/terra-form-textarea/CHANGELOG.md:4]: We don't need to talk about examples outside of the docs CHANGELOG.
```suggestion
* Added
  * Added support for `aria-invalid` through the `IsInvalid` prop to mark what field is invalid for assistive technology.
```

**Review by sdadn — APPROVED** [review]

File: packages/terra-form-textarea/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # Changelog
 
 ## Unreleased
+* Added
+  * Added support for `aria-invalid` through the `IsInvalid` prop to mark what field is invalid for assistive technology.
```
Comment (sycombs) [inline packages/terra-form-textarea/CHANGELOG.md:5]: nit: capitalization of the prop name
```suggestion
  * Added support for `aria-invalid` through the `isInvalid` prop to mark what field is invalid for assistive technology.
```

File: packages/terra-form-textarea/src/Textarea.jsx [hunk 3]
```diff
@@ -242,6 +242,7 @@ class Textarea extends React.Component {
     }
 
     additionalTextareaProps['aria-label'] = ariaLabelText;
+    additionalTextareaProps['aria-invalid'] = isInvalid;
```
Comment (sycombs) [inline packages/terra-form-textarea/src/Textarea.jsx:245]: Can we add a Jest test case for the value of the 'aria-invalid' when `isInvalid` is true?

**Review by sycombs — CHANGES_REQUESTED** [review]

**Review by sdadn — APPROVED** [review]

**Review by supreethmr — COMMENTED** [review]
Just curious we other input field components like terra-input-field and form-select-field component and they do communicate invalid state without `aria-invalid` attribute. so are we adding this attribute to fix any specific screen-reader browser combination OR adding it as part of best practice..??

**Comment by razvanluca** [comment]
> Just curious we other input field components like terra-input-field and form-select-field component and they do communicate invalid state without `aria-invalid` attribute. so are we adding this attribute to fix any specific screen-reader browser combination OR adding it as part of best practice..??

I did a test with Jaws screen reader and checked with @scottwilmarth what's the expected behavior of an invalid form-textarea and he told me about aria-invalid field. @scottwilmarth , do you want to add something on this topic?

**Comment by scottwilmarth** [comment]
Adding my +1 after reviewing example. I updated UX and accessibility labels.

**Comment by razvanluca** [comment]
I changed the initial Jira that was a spike to the current Epic, UXPLATFORM-9356 in the PR comment

**Review by supreethmr — APPROVED** [review]

**Pull request merged**